integration-testing/jpmsTest/src/debugDynamicAgentJpmsTest/kotlin/DynamicAttachDebugJpmsTest.kt

```diff
@@ -0,0 +1,46 @@
+@file:OptIn(ExperimentalCoroutinesApi::class)
+
+import org.junit.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.debug.*
+import org.junit.Ignore
+import org.junit.Test
+import java.io.*
+import java.lang.IllegalStateException
+import kotlin.test.*
+
+class DynamicAttachDebugJpmsTest {
+
+    /**
+     * Using:
+     *
+     *     jvmArgs("--add-exports=kotlinx.coroutines.debug/kotlinx.coroutines.repackaged.net.bytebuddy=com.sun.jna")
+     *     jvmArgs("--add-exports=kotlinx.coroutines.debug/kotlinx.coroutines.repackaged.net.bytebuddy.agent=com.sun.jna")
```

Comment:
What does this mean?

Reply:
You see that this test is Ignored: I could not get JPMS and Dynamic Debug Probes to be happy with each other. I suspect that the shading of byte-buddy is a problem here. 

This feature was not working previously, but I thought it's nice to keep the test and comment the 
- conditions under which this was tested
- the issue we were facing

Reply:
Let me double-check my understanding: do you mean that, if these JVM runtime arguments are added, the test will pass, but without them, it won't? Why not introduce a separate source set where these arguments are passed, so that we know this workaround works reliably?

Also, there could be a separate test that shows that there still *is* this problem; something like `assertFailsWith<IllegalStateException> { test body }`. With the additional comment like "we want this test to fail", we'll be able to immediately learn if something changes and this usage becomes possible without workarounds.

Reply:
As far as I am aware, Titouan Bion is already working on finding ways of making this test pass!